heat/classification/knn.py

```diff
@@ -0,0 +1,50 @@
+import heat as ht
+
+
+class KNN(ht.ClassificationMixin, ht.BaseEstimator):
+    """
+    HeAT implementation of the K-Nearest-Neighbours Algorithm.
+
+    This algorithm predicts labels to data vectors by using an already labeled training dataset as reference.
+    The input vector to be predicted is compared to the training vectors by calculating the distance.
+    We use the euclidean distance, this can be expanded to other distance functions as well.
+    Then a majority vote of the k nearest (smallest distances) training vectors labels is chosen as prediction.
+
+    Parameters
+    ----------
+    x : ht.DNDarray
+        array of shape (n_samples,), required
+        The Training vectors
+    y : ht.DNDarray
+        array of shape (n_samples,), required
+        Labels for the training set
+    num_neighbours: int, required
+        number of neighbours to consider when choosing label
+    --------
```

Comment:
Would you mind adding a reference section and having the original knn paper cited here?

Reply:
Example can be found here: https://github.com/helmholtz-analytics/heat/blob/504-docstring-formatting/heat/cluster/kmeans.py